cmake/onnxruntime_python.cmake

```diff
@@ -74,6 +74,15 @@ if (onnxruntime_ENABLE_TRAINING)
   target_include_directories(onnxruntime_pybind11_state PRIVATE ${ORTTRAINING_ROOT})
 endif()
 
+# Disable pybind11 warning on macOS
+# TODO, remove this after switch to pybind11 2.6.0+
+if(APPLE)
+  # https://github.com/pybind/pybind11/pull/2522
+  target_compile_options(onnxruntime_pybind11_state PRIVATE "-Wno-range-loop-analysis")
+  # https://github.com/pybind/pybind11/pull/2294
+  target_compile_options(onnxruntime_pybind11_state PRIVATE "-Wno-unused-value")
+endif()
+
```

Comment:
Moved disable dev mode to disable certain warning flags here, however we cannot check xcode version here since it is possible we are not using xcode as Cmake generator

Decide to still keep the use_dev_mode() in build.py since it will make it clearer to see what disables the dev mode instead of burying it in the generate_build_tree()